### ISSUE TITLE
Clicking "last post" for a thread in a forum view leads to first unread when user is logged in and not a mod

### ISSUE THREAD

**Issue by lairdshaw** [lukasamd/MyBB-View_Unread_posts#15]
The solution is to replace 'lastpost' with 'newpost' in the conditional on line 177 of inc/plugins/unreadPosts.php.

**Comment by lukasamd** [comment]
Can you describe this issue precisely? I cant understand the problem - tested with mod and it works the same like on standard user.

**Comment by lairdshaw** [comment]
Sure. Here's a (hopefully foolproof) way to reproduce it:

1. Create a new user with default privileges. The point of this is to ensure that the first unread post for this user for every thread is the first post in the thread.
2. Log in as that user.
3. Go to the forum's index page. Click on one of the thread links under the right-most column labelled "Last Post".

Observed behaviour (if my steps to reproduce are not flawed): Instead of being taken to the last post in the thread, you are taken to the first unread post (which just happens to be the very first post in the thread).

If you repeat this as an admin (making sure for testing purposes that on the given thread you test, you have unread posts prior to the last post), you will be correctly taken to the last post in the thread rather than the first unread.

I think that if you look at the fix I suggest, you will see that aside from fixing the problem (according to my testing), it is intuitively correct: the function it occurs in is named "actionNewpost", so, intuitively, line 177 should be checking against a new post rather than the last post.

**Comment by lukasamd** [comment]
@lairdshaw It isn't issue, but plugin assumption - users should be redirected to first unread posts if they are logged in. You can also disable this option in plugin settings on ACP.

**Comment by lairdshaw** [comment]
Ah, thanks for letting me know - I hadn't looked at the plugin's settings, my fault. Sorry for the mistaken bug report. Also, re-testing shows that you're right that there's no difference in behaviour between admin and ordinary accounts - perhaps I made a mistake when I tested the first time.

A suggestion: perhaps turn this feature off by default? It did confuse both the members of my forum as well as me: when you click on a link in a column titled "Last Post", you do expect to go to the last post.
